python/lsst/pipe/tasks/drpAssociationPipe.py

```diff
@@ -270,7 +270,9 @@ def visitDetectorPair(dataRef):
                                                    innerTractSkyRegion,
                                                    skyInfo.wcs)
                 associatedSsSources = associatedSsSources[ssInTractPatch]
-                associatedSsDiaSources = associatedSsDiaSources[ssInTractPatch]
+                assocDiaSrcIds = set(associatedSsSources['diaSourceId'])
+                diaSrcMask = [ID in assocDiaSrcIds for ID in associatedSsDiaSources['diaSourceId']]
```

Comment:
Nitpick: `ID` feels like a constant, and `id` is a built-in function. I'd go with something like `src_id` or `diaId` etc.